libyang/data.py

```diff
@@ -251,6 +252,41 @@ def new_meta(self, name: str, value: str, clear_dflt: bool = False):
         if ret != lib.LY_SUCCESS:
             raise self.context.error("cannot create meta")
 
+    def _get_attr_key(self, cdata_attr):
+        if cdata_attr.name.prefix != ffi.NULL:
+            return f"{c2str(cdata_attr.name.prefix)}:{c2str(cdata_attr.name.name)}"
+
+        return c2str(cdata_attr.name.name)
+
+    def attr(self):
+        ret = {}
+        for attr in self.cdata_attrs:
+            key = self._get_attr_key(attr)
+            ret[key] = c2str(attr.value)
+        return ret
+
+    def new_attr(self, name: str, value: str):
+        attrs = ffi.new("struct lyd_attr **")
+        ret = lib.lyd_new_attr(
+            self.cdata,
+            ffi.NULL,
+            str2c(name),
+            str2c(value),
+            attrs,
+        )
+        if ret != lib.LY_SUCCESS:
+            raise self.context.error("cannot create attr")
+        self.cdata_attrs.append(attrs[0])
+
+    def attr_free(self, name):
+        for attr in self.cdata_attrs:
+            key = self._get_attr_key(attr)
+
+            if name == key:
+                lib.lyd_free_attr_single(self.context.cdata, attr)
+                self.cdata_attrs.remove(attr)
+                break
```

Comment:
I wonder if we need a dedicated class for those attributes. @rjarry, what do you think?

Reply:
That may be overkill. These attributes don't make sense without a host `DNode`.

Reply:
OK.

Reply:
I sent my message too fast :D

Actually, yes. It would be better to have:

```python
dnode.attrs().get("foo")

for key, val in dnode.attrs():
    print(key, val)

dnode.attrs().set("foo", "bar")
dnode.attrs().remove("foo")
```

API stub:

```python
class DNodeAttrs:
    def get(self, name: str) -> str:
        ...
    def set(self, name: str, value: str):
        ...
    def remove(self, name: str):
        ...
    def __contains__(self, name: str):
        ...
    def __iter__(self):
        ...
    def __len__(self):
        ...

class DNode:
    def attrs(self) -> DNodeAttrs:
        ...
```

What do you think?

Reply:
Yes, I had something like that in mind. Agreed.

Reply:
I implemented your API suggestion